en/lessons/specifics/benchmarking.md

```diff
@@ -0,0 +1,346 @@
+Estimated total run time: 2.10 min
+
+
+
+Benchmarking with input large list:
+Benchmarking flat_map...
+
+Benchmarking with input medium list:
+Benchmarking flat_map...
+
+Benchmarking with input small list:
+Benchmarking flat_map...
+
+Benchmarking with input large list:
+Benchmarking map.flatten...
+
+Benchmarking with input medium list:
+Benchmarking map.flatten...
+
+Benchmarking with input small list:
+Benchmarking map.flatten...
+
+##### With input large list #####
+Name                  ips        average  deviation         median
+flat_map             6.29      158.93 ms    ±19.87%      160.19 ms
+map.flatten          4.80      208.20 ms    ±23.89%      200.11 ms
+
+Comparison:
+flat_map             6.29
+map.flatten          4.80 - 1.31x slower
+
+##### With input medium list #####
+Name                  ips        average  deviation         median
+flat_map           1.34 K        0.75 ms    ±28.14%        0.65 ms
+map.flatten        0.87 K        1.15 ms    ±57.91%        1.04 ms
+
+Comparison:
+flat_map           1.34 K
+map.flatten        0.87 K - 1.55x slower
+
+##### With input small list #####
+Name                  ips        average  deviation         median
+flat_map         122.71 K        8.15 μs   ±378.78%        7.00 μs
+map.flatten       86.39 K       11.58 μs   ±680.56%       10.00 μs
+
+Comparison:
+flat_map         122.71 K
+map.flatten       86.39 K - 1.42x slower
+```
+
+We can now see information for our benchmarks, grouped by input. This simple
+example doesn't provide any mind blowing insights, but you'd be very surprised
+how much performance varies based on input size!
+
+# Formatters
+
+The console output that we've seen is a helpful beginning for measuring the
+runtimes of your functions, but it's not your only option! In this section we'll
+look briefly at the three other available formatters, and also touch on what
+you'd need to do to write your own formatter if you like.
+
+## Other formatters
+
+Benchee has a console formatter built in, which is what we've seen already, but
+there are three other officially supported formatters - `benchee_csv`,
+`benchee_json` and `benchee_html`. Each of them does exactly what you would
+expect, which is writing the results to the named file formats so you can work
+with your results further in whichever format you like.
+
+Each of these formatters is a separate package, so to use them you need to add
+them as dependencies to your `mix.exs` file like so:
+
+```elixir
+defp deps do
+  [
+    {:benchee_csv,  "~> 0.6", only: :dev},
+    {:benchee_json, "~> 0.3", only: :dev},
+    {:benchee_html, "~> 0.3", only: :dev},
+  ]
+end
+```
+
+While `benchee_json` and `benchee_csv` are very simple, `benchee_html` is
+actually _very_ full featured! It can help you produce nice graphs and charts of
+from your results easily, and you can even export them as PNG images. All three
+formatters are well documented on their respective GitHub pages, so we won't
```

Comment:
I am not a native speaker, but shouldn't it be `well-documented`? If so, it should also be checked on [line 291](https://github.com/elixirschool/elixirschool/pull/1151/files#diff-5364d6d75e79d566801405a1097e95b8R291).